hudi-hadoop-common/src/main/java/org/apache/hudi/storage/hadoop/HoodieHadoopStorage.java

```diff
@@ -43,16 +47,66 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import static org.apache.hudi.common.util.ValidationUtils.checkArgument;
 import static org.apache.hudi.hadoop.fs.HadoopFSUtils.convertToHadoopPath;
 import static org.apache.hudi.hadoop.fs.HadoopFSUtils.convertToStoragePath;
 import static org.apache.hudi.hadoop.fs.HadoopFSUtils.convertToStoragePathInfo;
+import static org.apache.hudi.hadoop.fs.HadoopFSUtils.getFs;
 
 /**
  * Implementation of {@link HoodieStorage} using Hadoop's {@link FileSystem}
  */
 public class HoodieHadoopStorage extends HoodieStorage {
   private final FileSystem fs;
 
+  public HoodieHadoopStorage(HoodieStorage storage) {
```

Comment:
I feel like there are too many constructors here.  Ideally, only a couple should suffice.  To load the implementation through config and reflection, we cannot support many constructors, ideally only one or two
.

Reply:
Yeah, the reason for this is there are like 10 different HadoopFSUtils.getFs methods. I wanted to minimize the number of changes in this pr. We can follow up in another pr and reduce this.

Reply:
@jonvex have you created a JIRA for this one?